src/equations/equations.jl

```diff
@@ -87,6 +87,7 @@ end
 
 @inline Base.ndims(::AbstractEquations{NDIMS}) where {NDIMS} = NDIMS
 
+# TODO: @ranocha may this be removed?
```

Comment:
We need some setting - either `Ref(equations)` or the current solution.

Reply:
Thanks for the clarification. Unless you think it would be better to stick to the current approach, I suggest to use `Ref(equations)` since that's the [official proposed solution](https://docs.julialang.org/en/v1/manual/interfaces/#man-interfaces-broadcasting).

Reply:
You can go ahead with this change if you check the issues described in the comments below.

Reply:
Thanks to your excellent MWE, I can confirm that Julia v1.10 with all variantes of `--check-bounds` (no/yes/auto/nothing) yields zero allocations:
```julia
julia> struct BroadcastRef end

julia> Base.broadcastable(b::BroadcastRef) = Ref(b)

julia> struct BroadcastTuple end

julia> Base.broadcastable(b::BroadcastTuple) = (b,)

julia> foo(a, _) = a
foo (generic function with 1 method)

julia> function test(arg)
           n = 4
           dst = rand(n, n)
           src = rand(n, n)
           @allocated begin
               @. dst = foo(src, arg)
           end
       end
test (generic function with 1 method)

julia> test(BroadcastRef()), test(BroadcastTuple())
(0, 0)
```

I will thus go ahead and implement the suggested variant

Reply:
Done in 76386593c